app/models/poll.rb

```diff
@@ -40,11 +44,14 @@ def draft!
     super
   end
 
-  def vote!(preferences)
-    vote_results << preferences
-    vote = SchulzeBasic.do vote_results, vote_results.first.count
-    self.current_state = vote.ranks
-    save!
+  def vote!(user:, preferences:)
```

Comment:
Does it make sense here to use kwargs, if all parameters are non-optional, and there are only two of them? Your method calls will be so much shorter if you will make those regular arguments (but I'm open for discussion).